server/lib/csp.js

```diff
@@ -2,30 +2,41 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-// middleware to take care of CSP. CSP headers are only sent in development
-// mode, and are not sent if the user is running the unit tests.
+// Middleware to take care of CSP. CSP headers are not sent unless
+// config option 'cspEnabled' is set, with a special exception for
+// the /tests/index.html path, which are the frontend unit tests.
 
 'use strict';
 
 var helmet = require('helmet');
 var config = require('./configuration');
 
+var SELF = "'self'";
+var DATA = 'data:';
+var GRAVATAR = 'https://secure.gravatar.com';
+
 var cspMiddleware = helmet.csp({
-                      'default-src': ['\'self\''],
-                      'connect-src': [
-                        '\'self\'',
-                        config.get('fxaccount_url'),
-                        config.get('oauth_url'),
-                        config.get('profile_url')
-                      ],
-                      'img-src': [
-                        '\'self\'',
-                        'data:',
-                        config.get('profile_images_url'),
-                        'https://secure.gravatar.com'
-                      ],
-                      'report-uri': '/_/csp-violation'
-                    });
+  'default-src': [
+    SELF
+  ],
+
+  'connect-src': [
+    SELF,
+    config.get('fxaccount_url'),
+    config.get('oauth_url'),
+    config.get('profile_url')
+  ],
+
+  'img-src': [
+    SELF,
+    DATA,
+    GRAVATAR,
+    config.get('profile_images_url')
+  ],
+
+  'report-uri': config.get('cspReportUri'),
+  reportOnly: config.get('cspReportOnly')
```

Comment:
I was going to ask why `reportOnly` instead of `report-only`, but according to https://github.com/helmetjs/csp:

>  _"You can specify keys in a camel-cased fashion (`imgSrc`) or dashed (`img-src`); they are equivalent."_

That said, if we wanted to avoid having to quote the keys, we could probably switch all of them to camelCased. For example, `defaultSrc`, `connectSrc`, `imgSrc`, `reportUri`. It'd save us a whopping 12 characters. #performance